Travis

WeinaJi opened the pull request.

pramodk committed:
Fix travis.yml to correctly detect failures (#314)

Commit:
travis CI : add build for mac osx

Commit:
travis CI : add pyenv to path for taking the correct python version

ramcdougal committed:
updated nrnunits.lib.in and rxd.constants to 2019 SI values when --enable-LegacyFR=no (#323)

nrnhines committed:
src/nrnoc/nrnversion.h changes if 'git describe' changes. (#325)

pramodk committed:
Uniform formatting for CMake files (#327)

* Uniform formatting for CMake files

- Newly added cmake code / files are now formatted using cmake-format
package
- coding-convention submodule is added but there is still some work
needed to avoid formatting of comment blocks (issue opened in BBP
repository)

* Add submodule and initialize it if formatting option is added

* Disable recusive clonning

pramodk committed:
Cleanup and cmake formatting (#328)

- use cmake format fence ~~~ to avoid comment block formatting
 - only modules directory not formated properly otherwise one can do
   cmake .. -DNEURON_CMAKE_FORMAT=ON
   make cmake-format
   git checkout ../cmake/modules/    # undo formatting of cmake modules

pramodk commented:
@WeinaJi  : can you sync with @ohm314 about the build failures on OSX?

(make sure to also create a placeholder issue in jira for the task)

Commit:
travis: revert change

Commit:
travis ci: pick the correct python version

pramodk committed:
Doxygen integration (#321)

* An initial commit to get code documentation into NEURON

* Added cmake target and generate docs inside build dir

* A more reasonable logo

* Added some cmake instructions into README.md

* Reviewed all assert for necessary side effects. (#306) (#307)

30 converted from assert to nrn_assert. So work properly with NDEBUG
defined.

* Just minor cleanup

formatting in doxygen still doens't look great, but we can iron this out
later

* Added building doxygen in travis and deploying docs

* make doxygen only

* travis.yml fix

* fixed travis script to build doc for one job

* Added documentation preperation

* Reverted jobs changes for testing

* Small fix in README.md cmake options

* Added cmake option NRN_BUILD_DOC and selectively build doc in travis

* Reverted make install in travis

* Remove build option for doxygen

* Added some more groups

* update travis for docs

* Revert travis change

* Small fix to travis plan

* Small fix in if statement to build doc

pramodk committed:
Build python modules during make phase and copy them during install phase (#331)

* WIP : Prepare python modules during build time

* Introduce failure to validate travis CI

* add dependencies of nrnpython<ver> to hoc_module

* Fixed all tests

* Fix travis.yml to correctly detect failures

* Additional changes for build time cmake module

* Avoid setting rpath of build library : it was causing quite confusion
and failures from dlopen. We will use LD_LIBRARY_PATH or
DYLD_LIBRARY_PATH mechanism. From build directory we will use:

 NEURONHOME=`pwd`/share/nrn/  \
 PYTHONPATH=`pwd`/lib/python \
 python3.6 -c "from neuron import h; import neuron; neuron.test()"

* Avoid rpath to build but add it to libdir for linking

* Try to test from build directory

* Cleanup and cmake formatting
 - use cmake format fence ~~~ to avoid comment block formatting
 - only modules directory not formated properly otherwise one can do
   cmake .. -DNEURON_CMAKE_FORMAT=ON
   make cmake-format
   git checkout ../cmake/modules/    # undo formatting of cmake modules

* Add debug message for travis

* fix share/lib directory path

Commit:
travis CI: remove pyenv

pramodk committed:
Fixed issue with rxd tests from #313 (#332)

- constants.py was not copied to lib/python/neuron rather than
   share/python/neuron
 - make sure rxd test fails if import fails

Commit:
travis CI test with linux and macos

Commit:
travis CI: merge with upstream/cmake

Commit:
Travis CI: add test for mac os, merge with upstream/cmake
